app/src/main/java/org/systers/mentorship/remote/TokenAuthenticator.kt

```diff
@@ -18,9 +18,15 @@ class TokenAuthenticator: Authenticator{
     override fun authenticate(route: Route, response: Response): Request? {
         if (response.code() == 401) {
             preferenceManager.clear()
+
+            if(response.request().url().encodedPath() == "/login" || response.request().url().encodedPath() == "/register")
```

Comment:
Why have you added the register endpoint here?

Reply:
this is just to ensure nothing goes wrong in the signup activity in the future

Reply:
There is no 401 response on the register endpoint.